src/screens/ExchangeModal.js

```diff
@@ -420,7 +436,19 @@ export default function ExchangeModal({
         outputAmount,
         tradeDetails,
       };
-      await executeRap(wallet, type, swapParameters, callback);
+      let rapType = type;
+      switch (type) {
+        case ExchangeModalTypes.withdrawal:
+          rapType = RapActionTypes.withdrawCompound;
+          break;
+        case ExchangeModalTypes.deposit:
+          rapType = RapActionTypes.depositCompound;
+          break;
+        default:
+          rapType = RapActionTypes.swap;
+          break;
+      }
```

Comment:
wdyt of creating another method `getSwapRapTypeByExchangeType` containing this switch case so you can use it [here](https://github.com/rainbow-me/rainbow/pull/3613/files#diff-08d9008ab79a0765a3f4bca960079242e26c4a35125cf214a538f43497d9fd95R264)  as well